typespec-extension/src/code-model-builder.ts

```diff
@@ -548,121 +515,196 @@ export class CodeModelBuilder {
+          sharedApiVersions = apiVersions;
+        } else {
+          apiVersionSameForAllClients = isEqual(sharedApiVersions, apiVersions);
+        }
+        if (!apiVersionSameForAllClients) {
+          break;
+        }
       }
-      if (!apiVersionSameForAllClients) {
-        break;
+      if (apiVersionSameForAllClients) {
+        const serviceVersion = getServiceVersion(this.codeModel);
+        for (const client of this.codeModel.clients) {
+          client.serviceVersion = serviceVersion;
+        }
+      } else {
+        for (const client of this.codeModel.clients) {
+          const apiVersions = client.apiVersions;
+          if (apiVersions) {
+            client.serviceVersion = getServiceVersion(client);
+          }
+        }
       }
     }
-    if (apiVersionSameForAllClients) {
-      const serviceVersion = getServiceVersion(this.codeModel);
-      for (const client of this.codeModel.clients) {
-        client.serviceVersion = serviceVersion;
+  }
+
+  private buildSdkPathPathParameterForARM(): SdkPathParameter {
+    return {
+      kind: "path",
+      name: "endpoint",
+      isGeneratedName: true,
+      description: "Service host",
+      onClient: true,
+      urlEncode: false,
+      optional: false,
+      serializedName: "endpoint",
+      correspondingMethodParams: [],
+      type: {
+        kind: "string",
+        encode: "string",
+        decorators: [],
+        name: "string",
+        crossLanguageDefinitionId: "string",
+      },
+      isApiVersionParam: false,
+      decorators: [],
+      apiVersions: [],
+      crossLanguageDefinitionId: "endpoint",
+    };
+  }
+
+  private listSubClientsUnderClient(
+    client: SdkClientType<SdkHttpOperation>,
+    includeNestedOperationGroups: boolean,
+    isRootClient: boolean,
+  ): SdkClientType<SdkHttpOperation>[] {
+    const operationGroups: SdkClientType<SdkHttpOperation>[] = [];
+    for (const method of client.methods) {
+      if (method.kind === "clientaccessor") {
+        const subClient = method.response;
+        if (!isRootClient) {
+          // if it is not root client, append the parent client's name
+          subClient.name = this.removeClientSuffix(client.name) + this.removeClientSuffix(pascalCase(subClient.name));
+        }
+        operationGroups.push(subClient);
+        if (includeNestedOperationGroups) {
+          for (const operationGroup of this.listSubClientsUnderClient(subClient, includeNestedOperationGroups, false)) {
+            operationGroups.push(operationGroup);
+          }
+        }
       }
-    } else {
-      for (const client of this.codeModel.clients) {
-        const apiVersions = client.apiVersions;
-        if (apiVersions) {
-          client.serviceVersion = getServiceVersion(client);
+    }
+    return operationGroups;
+  }
+
+  private listServiceMethodsUnderClient(
+    client: SdkClientType<SdkHttpOperation>,
+    includeNestedServiceMethods: boolean,
```

Comment:
This `includeNestedServiceMethods` param seems always `false`? Any case you would pass `true`?

Reply:
No, we don't pass in `true` for now, just have the flag in case we need to do so. I can delete it.